go-controller/pkg/node/iptables/iptables_manager.go

```diff
@@ -0,0 +1,402 @@
+package iptables
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/util/iptables"
+	kexec "k8s.io/utils/exec"
+)
+
+type Chain struct {
+	Table iptables.Table
+	Chain iptables.Chain
+	Proto iptables.Protocol
+}
+
+func (c Chain) equal(c2 Chain) bool {
+	if c.Table != c2.Table {
```

Comment:
The struct doesn't have any pointers, why do we need the `equal` function?
isn't using `c == c2` easier?

Reply:
True. I didnt consider this was possible. Updated.